src/Workspaces/Core/Portable/LanguageServices/SyntaxFactsService/AbstractSyntaxFactsService.cs

```diff
@@ -542,11 +542,22 @@ public bool IsOnHeader<THoleSyntax>(int position, SyntaxNode ownerOfHeader, Synt
 
         protected int GetStartOfNodeExcludingAttributes(SyntaxNode node)
         {
-            var attributeLists = GetAttributeLists(node);
-            var start = attributeLists.LastOrDefault()?.GetLastToken().GetNextToken().SpanStart ??
-                        node.SpanStart;
+            var attributeList = GetAttributeLists(node);
+            if (attributeList.Any())
+            {
+                var endOfAttributeLists = attributeList.Last().Span.End;
```

Comment:
can you use Length/Count rather than linq? GetAttributeLists return SyntaxList so, we should be able to get those data without involving IEnumerable and iteration unless you know that Linq are using optimized version that uses index for SyntaxList for like Last()

Reply:
There's an optimized version for SyntaxList.